### PR TITLE
COMPAT: Matplotlib 3.5.0

### LINKED ISSUE #44521: CI: test failing in the plotting module
### 

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the master branch of pandas.

### Reproducible Example

```python
Create a PR the workflows will run and in the `Test experimental data manager (slow)` workflow it it will return an exit code 1
Launch `pytest pandas/tests/plotting/test_hist_method.py pandas/tests/plotting/test_series.py pandas/tests/plotting/frame/test_hist_box_by.py`

All tests used to pass a week or so ago, now it is failing (and failing in many different PRs, does not look like an isolate case).
```

### Issue Description

The following tests are consistently failing on local installation or in the CI/CD pipeline.

```
=========================== short test summary info ============================
FAILED pandas/tests/plotting/test_hist_method.py::TestDataFrameGroupByPlots::test_axis_share_x
FAILED pandas/tests/plotting/test_hist_method.py::TestDataFrameGroupByPlots::test_axis_share_y
FAILED pandas/tests/plotting/test_hist_method.py::TestDataFrameGroupByPlots::test_axis_share_xy
FAILED pandas/tests/plotting/test_series.py::TestSeriesPlots::test_area_sharey_dont_overwrite
FAILED pandas/tests/plotting/frame/test_frame.py::TestDataFramePlots::test_area_sharey_dont_overwrite
FAILED pandas/tests/plotting/frame/test_hist_box_by.py::TestHistWithBy::test_axis_share_x_with_by
FAILED pandas/tests/plotting/frame/test_hist_box_by.py::TestHistWithBy::test_axis_share_y_with_by
```

### Expected Behavior

Tests should pass

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit           : 8c4bd0e0c3698ced1420c41d33bac9f0e8fb4916
python           : 3.8.12.final.0
python-bits      : 64
OS               : Linux
OS-release       : 4.19.104-microsoft-standard
Version          : #1 SMP Wed Feb 19 06:37:35 UTC 2020
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : C.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 1.4.0.dev0+1144.g8c4bd0e0c3.dirty
numpy            : 1.21.4
pytz             : 2021.3
dateutil         : 2.8.2
pip              : 21.3.1
setuptools       : 59.1.1
Cython           : 0.29.24
pytest           : 6.2.5
hypothesis       : 6.24.6
sphinx           : 4.3.0
blosc            : None
feather          : None
xlsxwriter       : 3.0.2
lxml.etree       : 4.6.4
html5lib         : 1.1
pymysql          : None
psycopg2         : None
jinja2           : 3.0.3
IPython          : 7.29.0
pandas_datareader: None
bs4              : 4.10.0
bottleneck       : 1.3.2
fsspec           : 2021.11.0
fastparquet      : 0.7.1
gcsfs            : 2021.11.0
matplotlib       : 3.5.0
numexpr          : 2.7.3
odfpy            : None
openpyxl         : 3.0.9
pandas_gbq       : None
pyarrow          : 6.0.1
pyxlsb           : None
s3fs             : 2021.11.0
scipy            : 1.7.2
sqlalchemy       : 1.4.27
tables           : 3.6.1
tabulate         : 0.8.9
xarray           : 0.18.2
xlrd             : 2.0.1
xlwt             : 1.3.0
numba            : 0.53.1
</details>

### PR DESCRIPTION
**Pull request by lithomas1**
- [ ] closes #44521 
- [ ] tests added / passed
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them
- [ ] whatsnew entry

Still some deprecation warnings left. Anyone want to approve for me so we can get this merged ASAP?

### CODE REVIEW

**Comment by jreback** [comment]
we are fully testing i assume?

can u add. what's new note for 1.3.5

File: pandas/tests/plotting/common.py [hunk 1]
```diff
@@ -569,6 +571,12 @@ def _unpack_cycler(self, rcParams, field="color"):
         """
         return [v[field] for v in rcParams["axes.prop_cycle"]]
 
+    def get_x_axis(self, ax):
+        return ax._shared_axes["x"] if self.compat.mpl_ge_3_5_0() else ax._shared_x_axes
+
+    def get_y_axis(self, ax):
```
Comment (jbrockmendel) [inline pandas/tests/plotting/common.py:577]: could annotate ax?

File: pandas/plotting/_matplotlib/core.py [hunk 2]
```diff
@@ -1036,6 +1036,7 @@ def _plot_colorbar(self, ax: Axes, **kwds):
         # use the last one which contains the latest information
         # about the ax
         img = ax.collections[-1]
+        ax.grid(False)
```
Comment (jbrockmendel) [inline pandas/plotting/_matplotlib/core.py:1039]: what is this doing?

**Comment by jbrockmendel** [comment]
re-started the timed-out build.  one question about the ax.grid(False), otherwise LGTM pending green(ish)

**Comment by jbrockmendel** [comment]
gets us back to green on everything except the docbuild.  @jreback good to go?

**Comment by jreback** [comment]
yep merge when ready

**Comment by jbrockmendel** [comment]
thanks @lithomas1

**Comment by simonjayhawkins** [comment]
> can u add. what's new note for 1.3.5

did we want a release note for this?

**Pull request merged**